[Contraction] Compilation issue with CGAL 5.5

With the modification of `STL_Extension/include/CGAL/Modifiable_priority_queue.h` (cf. [https://github.com/CGAL/cgal/compare/v5.4.2...v5.5]()) in CGAL 5.5, GUDHI contraction module is no more compiling:
```
In file included from gudhi-devel/src/Contraction/include/gudhi/Edge_contraction.h:15,
                 from gudhi-devel/src/Contraction/example/Rips_contraction.cpp:10:
gudhi-devel/src/Contraction/include/gudhi/Skeleton_blocker_contractor.h: In instantiation of ‘class Gudhi::contraction::Skeleton_blocker_contractor<Gudhi::skeleton_blocker::Skeleton_blocker_geometric_complex<Gudhi::skeleton_blocker::Skeleton_blocker_simple_geometric_traits<Geometry_trait> > >’:
gudhi-devel/src/Contraction/example/Rips_contraction.cpp:65:33:   required from here
gudhi-devel/src/Contraction/include/gudhi/Skeleton_blocker_contractor.h:175:31: error: no type named ‘handle’ in ‘class CGAL::Modifiable_priority_queue<boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>, Gudhi::contraction::Skeleton_blocker_contractor<Gudhi::skeleton_blocker::Skeleton_blocker_geometric_complex<Gudhi::skeleton_blocker::Skeleton_blocker_simple_geometric_traits<Geometry_trait> > >::Compare_cost, Gudhi::contraction::Skeleton_blocker_contractor<Gudhi::skeleton_blocker::Skeleton_blocker_geometric_complex<Gudhi::skeleton_blocker::Skeleton_blocker_simple_geometric_traits<Geometry_trait> > >::Undirected_edge_id, CGAL::CGAL_BOOST_PENDING_MUTABLE_QUEUE>’
  175 |   typedef typename PQ::handle pq_handle;
      |                               ^~~~~~~~~
```

This may be a good opportunity to remove CGAL dependency on this module, only required because of the priority queue management.

--- a/src/Contraction/include/gudhi/Skeleton_blocker_contractor.h
+++ b/src/Contraction/include/gudhi/Skeleton_blocker_contractor.h
@@ -172,7 +172,7 @@ typename GeometricSimplifiableComplex::Vertex_handle> {
   };
 
   typedef CGAL::Modifiable_priority_queue<Edge_handle, Compare_cost, Undirected_edge_id> PQ;
-  typedef typename PQ::handle pq_handle;
+  typedef bool pq_handle;
 
 
   // An Edge_data is associated with EVERY edge in the complex (collapsible or not).
@@ -196,7 +196,7 @@ typename GeometricSimplifiableComplex::Vertex_handle> {
     }
 
     bool is_in_PQ() const {
-      return PQHandle_ != PQ::null_handle();
+      return PQHandle_ != false;
     }
 
     void set_PQ_handle(pq_handle h) {
@@ -204,7 +204,7 @@ typename GeometricSimplifiableComplex::Vertex_handle> {
     }
 
     void reset_PQ_handle() {
-      PQHandle_ = PQ::null_handle();
+      PQHandle_ = false;
     }
 
    private:
@@ -238,16 +238,22 @@ typename GeometricSimplifiableComplex::Vertex_handle> {
   }
 
   void insert_in_PQ(Edge_handle edge, Edge_data& data) {
-    data.set_PQ_handle(heap_PQ_->push(edge));
+    heap_PQ_->push(edge);
+    data.set_PQ_handle(true);
     ++current_num_edges_heap_;
   }
 
   void update_in_PQ(Edge_handle edge, Edge_data& data) {
+#if CGAL_VERSION_NR < 1050500000
     data.set_PQ_handle(heap_PQ_->update(edge, data.PQ_handle()));
+#else
+    heap_PQ_->update(edge);
+#endif
   }
 
   void remove_from_PQ(Edge_handle edge, Edge_data& data) {
-    data.set_PQ_handle(heap_PQ_->erase(edge, data.PQ_handle()));
+    heap_PQ_->erase(edge);
+    data.set_PQ_handle(false);
     --current_num_edges_heap_;
   }
 

--- a/src/Contraction/include/gudhi/Skeleton_blocker_contractor.h
+++ b/src/Contraction/include/gudhi/Skeleton_blocker_contractor.h
@@ -171,7 +171,12 @@ typename GeometricSimplifiableComplex::Vertex_handle> {
     Self const* algorithm_;
   };
 
+#if CGAL_VERSION_NR < 1050500000
   typedef CGAL::Modifiable_priority_queue<Edge_handle, Compare_cost, Undirected_edge_id> PQ;
+#else
+  typedef CGAL::Modifiable_priority_queue<Edge_handle, Compare_cost, Undirected_edge_id, CGAL::CGAL_BOOST_PENDING_RELAXED_HEAP> PQ;
+#endif
+  
   typedef bool pq_handle;
 
 

